newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -0,0 +1,285 @@
+import Window from '../../Utils/Window';
+import { Line, Spacer } from '../../UI/Grid';
+import CircularProgress from '../../UI/CircularProgress';
+import BackgroundText from '../../UI/BackgroundText';
+import { showErrorBox } from '../../UI/Messages/MessageBox';
+import Mark from '../../UI/CustomSvgIcons/Mark';
+import { AssetStoreContext } from '../AssetStoreContext';
+import FlatButton from '../../UI/FlatButton';
+import { LineStackLayout } from '../../UI/Layout';
+
+type Props = {|
+  privateAssetPackListingData: PrivateAssetPackListingData,
+  onSuccessfulPurchase: () => Promise<void>,
+  onClose: () => void,
+|};
+
+const PrivateAssetPackPurchaseDialog = ({
+  privateAssetPackListingData,
+  onClose,
+  onSuccessfulPurchase,
+}: Props) => {
+  const {
+    profile,
+    getAuthorizationHeader,
+    onLogin,
+    onCreateAccount,
+    receivedAssetPacks,
+  } = React.useContext(AuthenticatedUserContext);
+  const { loadedReceivedAssetPackInStore } = React.useContext(
+    AssetStoreContext
+  );
+  const [isPurchasing, setIsPurchasing] = React.useState(false);
+  const [
+    isCheckingPurchasesAfterLogin,
+    setIsCheckingPurchasesAfterLogin,
+  ] = React.useState(!receivedAssetPacks);
+  const [purchaseSuccessful, setPurchaseSuccessful] = React.useState(false);
+  const onStartPurchase = async () => {
+    if (!profile) return;
+    try {
+      setIsPurchasing(true);
+      const checkoutUrl = await getStripeCheckoutUrl(getAuthorizationHeader, {
+        stripePriceId: privateAssetPackListingData.prices[0].stripePriceId,
+        userId: profile.id,
+        customerEmail: profile.email,
+      });
+      Window.openExternalURL(checkoutUrl);
+    } catch (error) {
+      console.error('Unable to get the checkout URL', error);
+      setIsPurchasing(false);
+      showErrorBox({
+        message: `Unable to get the checkout URL. Please try again later.`,
+        rawError: error,
+        errorId: 'asset-pack-checkout-error',
+      });
+    }
+  };
+
+  const checkUserPurchases = React.useCallback(
+    async () => {
+      if (!profile) return;
+      try {
+        const userPurchases = await listUserPurchases(getAuthorizationHeader, {
+          userId: profile.id,
+          productType: 'asset-pack',
+          role: 'receiver',
+        });
+        if (
+          userPurchases.find(
+            userPurchase =>
+              userPurchase.productId === privateAssetPackListingData.id
+          )
+        ) {
+          // We found the purchase, the user has bought the asset pack.
+          // We do not close the dialog yet, as we need to trigger a refresh of the asset store.
+          await onSuccessfulPurchase();
+        }
+      } catch (error) {
+        console.error('Unable to get the user purchases', error);
+        showErrorBox({
+          message: `An error happened while checking if your purchase was successful. If you have finished the payment, refresh the app to see your asset pack!`,
```

Comment:
I'm think about `completed the payment` but I'm not sure it's better.

Reply:
I'm not sure one can refresh the app on Electron.
Would close and re-open the asset store work?